### ISSUE TITLE
What does that mean " update site source control" failed with System.net.WebException?

### ISSUE THREAD

**Issue by tyi1473** [OfficeDev/microsoft-teams-apps-company-communicator#1018]
Repository 'UpdateSiteSourceControl' operation failed with System.Net.WebException: The remote server returned an error: (502) Bad Gateway. at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult) at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.Administration.Utilities.WebRequestHelper.<GetResponseAsyncWithTimeout>d__10.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.Administration.SiteRepositoryProvider.TrackerContext.<GetResponseAsync>d__41.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.Administration.SiteRepositoryProvider.<SetSettingsAsync>d__31.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.Administration.ExternalSiteRepositoryProvider.<UpdateSiteSourceControl>d__5.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.Administration.WebCloudController.<>c__DisplayClass348_1.<<UpdateSiteSourceControl>b__1>d.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Web.Hosting.AsyncHelper.RunSync[TResult](Func`1 func) at Microsoft.Web.Hosting.Administration.WebCloudController.UpdateSiteSourceControl(String subscriptionName, String webspaceName, String name, SiteSourceControl siteSourceControl). Click here for details

**Comment by gsv022** [comment]
Hi @tyi1473 ,
Thanks for raising the query. Normally, This issue occurs when the resource like App service or function app tries to pull the code from external git, sometimes it fails to pull the code due to network issue which is not in our control. This leads to 502 bad gateway error.

- To overcome the failed resources, Go to azure portal-->Go to resource group(Where the deployment has been done)-->Select  Data-Function from list of resources-->Go to Deployment Center-->Click on Settings

<img width="592" alt="image" src="https://user-images.githubusercontent.com/98143492/220826419-783f50cb-7868-492e-b35d-9392364b4b97.png">

- If the resource is not pointing to intended repository and  showing empty. Please connect to the external Git and point to the official repository.
- Once the status turns to Success(active) in the logs section then good to proceed with remaining steps of deployment.
Note  Do the same for all failed resources. If you are still facing the issue, to unblock yourself please follow the manual deployment guide.


Hope we conveyed.

**Comment by gsv022** [comment]
@tyi1473 , Once we connect the web app to the intended repository we can check the status in logs section of deployment center. If it is successful, you can proceed with the remaining steps of deployment. Same applies for function apps as well.
**Note:**- Once the status shows successful, no need to re-deploy the template again.

**Comment by tyi1473** [comment]
OK. even though status shows successful,  3 function apps have shown System Private.URi: Value cannot be null 
![FunctionApps_error2](https://user-images.githubusercontent.com/65746803/233731358-b70c60f0-2f9e-4af6-9e3b-0c28e744fe71.PNG)
![FunctionApps_error](https://user-images.githubusercontent.com/65746803/233731370-18ce9137-b1b1-40b5-a86c-d7ed7ef5f882.PNG)

**Comment by v-royavinash** [comment]
@tyi1473, It appears that during deployment, all other resources except for the web apps and functions app failed. Could you please share the complete deployment logs to understand which resources failed? 

Please note that if any other resources failed during deployment, apart from the three function apps and app service, you will need to redeploy the resources from the beginning.

**Comment by tyi1473** [comment]
I couldn't reproduce the same instance from my work network. But I can tell the issues  from marole/web or resouceName/web will keep happening from my work or my home network. 
![ExternalGit_conflict](https://user-images.githubusercontent.com/65746803/235213585-e8ca56e1-22be-444f-98e7-38de342fbc97.PNG)
That said " Message
There was a conflict. Conflict with existing ScmType: ExternalGit"

So as long as you told me what to do, I think I am good.  Source control conflict  was caused by our subscription or something else ?  
Unusual case : another smart detector alert rule was created " Failure Anomalies " 
![image](https://user-images.githubusercontent.com/65746803/235214871-ca7f17ba-4656-4eae-b0e9-97b2032c127f.png)
Failure Anomalies notifies you of an unusual rise in the rate of failed HTTP requests or dependency calls.

I think "resouceName/web" creates high volume of transactions so that application deployment was stopped and throwed the error.

**Comment by gsv022** [comment]
@tyi1473 ,Please share the screenshot of deployment status of App service(web app). Follow the same procedure as mentioned below.

**Comment by gsv022** [comment]
Hi @tyi1473 ,
Please try the below procedure for app service failure
- Go to azure portal-->Go to resource group(Where the deployment has been done)-->Select  App Service from list of resources
- Go to configuration section-->Add the below parameter.
**WEBSITE_NPM_DEFAULT_VERSION: 8.19.2**

- Save it
- Then go to deployment center-->Do SYNC again.

Hope it helps.

**Comment by tyi1473** [comment]
@gsv022,
I did what you told me. I am more concerning about what makes me failed to deploy web app in the first time. 
![image](https://user-images.githubusercontent.com/65746803/236860001-f87110a5-f730-4d30-b255-7a21a2b5e022.png)

Whenever I run the deployment, the deployment failed me every time. Is it our network issue or your network?

**Comment by gsv022** [comment]
Hi @tyi1473 ,
Could you please share the App service deployment status screenshot as shown below. 

> Hi @tyi1473 , Thanks for raising the query. Normally, This issue occurs when the resource like App service or function app tries to pull the code from external git, sometimes it fails to pull the code due to network issue which is not in our control. This leads to 502 bad gateway error.
> 
> * To overcome the failed resources, Go to azure portal-->Go to resource group(Where the deployment has been done)-->Select  Data-Function from list of resources-->Go to Deployment Center-->Click on Settings
> 
> <img alt="image" width="592" src="https://user-images.githubusercontent.com/98143492/220826419-783f50cb-7868-492e-b35d-9392364b4b97.png">
> 
> * If the resource is not pointing to intended repository and  showing empty. Please connect to the external Git and point to the official repository.
> * Once the status turns to Success(active) in the logs section then good to proceed with remaining steps of deployment.
>   Note  Do the same for all failed resources. If you are still facing the issue, to unblock yourself please follow the manual deployment guide.
> 
> Hope we conveyed.

**Comment by tyi1473** [comment]
OK thanks for your explanation.  the log I found from the PowerShell deployment told. 
![image](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/65746803/b89ebe81-aa43-462f-b97e-ad42f3c27afc)
Doesn't it have something to do with permission issue? 
That deployment threw the errors like these.
![image](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/65746803/36b817b8-5933-4ea8-a39f-203bcc2d6bfa)

**Comment by tyi1473** [comment]
I deleted the previous Resources Group. I ran the same deployment to another subscription which was successfully deployed.
![image](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/65746803/9cf2e811-e9a2-4cee-9f33-e4567267a258)

**Comment by gsv022** [comment]
@tyi1473, Can't comment on it. As you mentioned, it is a new deployment, the deployment seems to be succeeded. Are you still facing the same issue?

**Comment by tyi1473** [comment]
Yes. Whenever I deleted the failed deployment and run the same thing from the PowerShell deployment on previous subscription, the same issues occurred again and again. No permanent soultion.

**Comment by gsv022** [comment]
@tyi1473 , Please share the deployment status of App service(web app) by following the above steps. It is same as prep-function that you shared.

**Comment by tyi1473** [comment]
Sorry. as I mentioned earlier, the deployment was cured and deleted to find out to have the same occurrence when I repeatedly run the same deployment. So sync option is temporary fix and it is working but I want to know what is causing that same problem.

I have a contributor role that was given by the tenant administration. I was very suspicious of getting system identity role that the administration gave.

**Comment by gsv022** [comment]
@tyi1473 ,
As this deployment is long process it will take good amount of time and its completely not in our control. To mitigate the risks associated with lengthy deployments, we can employ a strategic approach that enhances our control over the process. By breaking down the template into smaller, more manageable chunks and deploying them individually, we increase our chances of success. 

Hope we conveyed

**Comment by gsv022** [comment]
We are closing this request as we have provided the inputs for the query raised. If you face any issue with company communicator please raise a new issue for further assistance.
